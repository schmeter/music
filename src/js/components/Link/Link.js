import React from 'react';
import PropTypes from 'prop-types';
import { NavLink } from 'react-router-dom';


const isExternal = (path) => /^https?:\/\/.+/.test(path);

class Link extends React.Component {
    render() {
        const { className, to, onClick, children } = this.props;
        const props = {
            className,
            onClick
        };
        const content = children || to;
        return isExternal(to) ? (
            <a
                target="_blank"
                href={to}
                {...props}
            >
                {content}
            </a>
        ) : (
            <NavLink
                exact
                activeClassName="active"
                to={to}
                {...props}
            >
                {content}
            </NavLink>
        );
    }
}

Link.propTypes = {
    to: PropTypes.string.isRequired,
    children: PropTypes.node,
    onClick: PropTypes.func,
    className: PropTypes.string
};

export default Link;
