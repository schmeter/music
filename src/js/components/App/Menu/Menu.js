import React from 'react';
import PropTypes from 'prop-types';

import Link from '../../Link';
import Icon from '../../Icon';
import i18n from '../../../services/i18n';
import { getUrl } from '../../../services/navigation';

const Menu = ({
    indexRoute,
    closeLayers,
}) => {
    const routes = [
        'audio',
        'video',
        'event',
        'settings',
    ];

    return (
        <ul className="layer-content">
            {routes.map(route => (
                <li
                    key={route}
                    className="link"
                >
                    <Icon id="hand-o-right" />
                &nbsp;
                    <Link
                        to={getUrl(indexRoute === route ? 'index' : route)}
                        onClick={closeLayers}
                    >
                        {i18n(`page_${route}_title`)}
                    </Link>
                </li>
            ))}
        </ul>
    );
};

Menu.propTypes = {
    indexRoute: PropTypes.string,
    closeLayers: PropTypes.func.isRequired,
};

export default Menu;
