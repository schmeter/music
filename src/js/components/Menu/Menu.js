import React from 'react';
import PropTypes from 'prop-types';

import Link from '../Link';
import Icon from '../Icon';
import i18n from '../../services/i18n';
import { getRouteWithParams } from '../../services/navigation';

const Menu = ({
  full,
  closeLayers,
}) => {
  const routes = [
    'audio',
    'video',
    'event',
    full && 'tools',
    full && 'settings',
  ].filter(route => !!route);

  return (
    <ul className="menu">
      {routes.map((route, index) => (
        <li
          key={route}
          className="link"
        >
          <Icon id="hand-o-right" />
          &nbsp;
          <Link
            to={getRouteWithParams(route)}
            onClick={closeLayers}
          >
            {i18n(`page_${route}_title`)}
          </Link>
        </li>
      ))}
    </ul>
  );
};

Menu.propTypes = {
  full: PropTypes.bool,
  closeLayers: PropTypes.func.isRequired,
};

export default Menu;
