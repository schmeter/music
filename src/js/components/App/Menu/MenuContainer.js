import { connect } from 'react-redux';

import Menu from './Menu';
import { closeLayersAction } from '../../../state/actions/layer';

export const mapDispatchToProps = dispatch => ({
    closeLayers: () => dispatch(closeLayersAction()),
});

export default connect(null, mapDispatchToProps)(Menu);
