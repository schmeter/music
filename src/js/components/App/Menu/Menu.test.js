import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Menu from './Menu';
import Link from '../../Link';

describe('Menu', () => {
    const mockedEvent = {
        preventDefault: jest.fn(),
    };

    it('renders correctly', () => {
        const props = {
            indexRoute: 'audio',
            closeLayers: jest.fn(),
        };
        const component = shallow(<Menu {...props} />);

        expect(toJson(component)).toMatchSnapshot();

        component.find(Link).first().simulate('click', mockedEvent);

        expect(props.closeLayers).toHaveBeenCalled();
    });
});
