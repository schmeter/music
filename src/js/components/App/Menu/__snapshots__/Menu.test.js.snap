// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu renders correctly 1`] = `
<ul
  className="layer-content"
>
  <li
    className="link"
    key="audio"
  >
    <Icon
      id="hand-o-right"
    />
     
    <Link
      onClick={[MockFunction]}
      to="/audio"
    >
      Alben
    </Link>
  </li>
  <li
    className="link"
    key="video"
  >
    <Icon
      id="hand-o-right"
    />
     
    <Link
      onClick={[MockFunction]}
      to="/video"
    >
      Videos
    </Link>
  </li>
  <li
    className="link"
    key="event"
  >
    <Icon
      id="hand-o-right"
    />
     
    <Link
      onClick={[MockFunction]}
      to="/event"
    >
      Gigs
    </Link>
  </li>
  <li
    className="link"
    key="tools"
  >
    <Icon
      id="hand-o-right"
    />
     
    <Link
      onClick={[MockFunction]}
      to="/tools"
    >
      Hilfe
    </Link>
  </li>
  <li
    className="link"
    key="settings"
  >
    <Icon
      id="hand-o-right"
    />
     
    <Link
      onClick={[MockFunction]}
      to="/settings"
    >
      Einstellungen
    </Link>
  </li>
</ul>
`;
